result.go

```diff
@@ -31,6 +32,10 @@ type Table interface {
 	// The function f will be called zero or more times.
 	Do(f func(ColReader) error) error
 
+	// DoArrow calls f to process the data contained within the table.
+	// It uses the arrow buffers.
+	DoArrow(f func(ArrowColReader) error) error
```

Comment:
This design looks good. The idea being that nothing changes right now and then we slowly move each transformation over to using this method. Once they are all complete we can then remove `Do` and `ColReader` and rename `DoArrow` and `ArrowColReader` to simply `Do` and `ColReader`.